variables.tf

```diff
@@ -0,0 +1,82 @@
+# ------------------------------------------------------------------------------
+# Variables
+# ------------------------------------------------------------------------------
+variable "name_prefix" {
+  description = "A prefix used for naming resources."
+}
+
+variable "vpc_id" {
+  description = "The ID of the VPC that this container will run in, needed for the Target Group"
+}
+
+variable "private_subnet_ids" {
```

Comment:
This is labelled private_subnet_ids - but in the example public subnets are provided here. And the later variable for assigning public IP addresses would only apply if the subnet was public.  I believe this module should either be opinionated and only work in private subnets (remove the public IP address too) or this variable should be renamed.

Reply:
This is done in other modules as well, and I think it's fine because `private_subnet_ids` strongly suggests that the subnets should be private, and the module will create any resources that _should_ be private in these subnets - but ultimately that is left up to the user. For examples it is nice to not have to create a new VPC and just pass in the public subnets of the default VPC for testing purposes.

As for assigning public IP addresses, I think they added it to be able to use the default VPC in tests - but I don't think we have a use case for this variable since Fargate containers need to be behind a load balancer anyway?

Reply:
I do agree that this should be private as best practice but to be consistent with the other examples and use the default VPC we need to have a public IP (since the default VPC doesn't have NAT gateways and to access internet through the internt gateway you need public IP). We could clarify this in the documentation/description for the public ip variable. The other option would be to rewrite the other examples for ECS to not use the default VPC (think we should be consistent here)